backend/siarnaq/api/compete/views.py

```diff
@@ -315,6 +316,59 @@ def scrimmage(self, request, pk=None, *, episode_id):
         serializer = self.get_serializer(page, many=True)
         return self.get_paginated_response(serializer.data)
 
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="team_id",
+                type=int,
+                description="A team to filter for. Defaults to your own team.",
+            ),
+        ],
+        responses={
+            status.HTTP_204_NO_CONTENT: "No ranked matches found.",
+            status.HTTP_200_OK: HistoricalRatingSerializer(many=True),
+        },
+    )
+    @action(
+        detail=False,
+        methods=["get"],
+        permission_classes=(IsEpisodeMutable,),
+    )
+    def historical_ranking(self, request, pk=None, *, episode_id):
+        """List the historical rating of a team."""
+        queryset = Match.objects.all().filter(tournament_round__isnull=True)
+
+        team_id = parse_int(self.request.query_params.get("team_id"))
+        if team_id is not None:
+            queryset = queryset.filter(participants__team=team_id)
+        else:
+            queryset = queryset.filter(participants__team__members=request.user.pk)
+        has_invisible = self.get_queryset().filter(
+            participants__team__status=TeamStatus.INVISIBLE
+        )
+        queryset = queryset.exclude(pk__in=Subquery(has_invisible.values("pk")))
+        queryset = queryset.filter(is_ranked=True)
+
+        if queryset.all().count() == 0:
+            return Response(None, status=status.HTTP_204_NO_CONTENT)
+        else:
+            matches = queryset.order_by("created")
+
+        ordered = [
+            {
+                "timestamp": match.created,
+                "rating": match.participants.get(team=team_id).rating
+                if team_id is not None
+                else match.participants.get(team__members__pk=request.user.pk).rating,
+            }
+            for match in matches
+        ]
+
+        results = HistoricalRatingSerializer(
+            self.paginate_queryset(ordered), many=True
```

Comment:
I don't think we need to paginate here

Reply:
gaurav's right, this in fact _needs_ to not be paginated -- this was the core problem that turned this into an issue+PR in the first place